tests/test_functions.py

```diff
@@ -106,3 +117,45 @@ def test_convert_excel_date():
 def test_fill_empty(null_df):
     df = jn.DataFrame(null_df).fill_empty(columns=['2'], value=3)
     assert set(df.loc[:, '2']) == set([3])
+
+
+def test_multiindex_clean_names_functional(multiindex_dataframe):
+    df = clean_names(multiindex_dataframe)
+
+    levels = [
+        ['a', 'bell_chart', 'decorated_elephant'],
+        ['b', 'normal_distribution', 'r_i_p_rhino_']
+    ]
+
+    labels = [[1, 0, 2], [1, 0, 2]]
+
+    expected_columns = pd.MultiIndex(levels=levels, labels=labels)
+    assert set(df.columns) == set(expected_columns)
+
+
+def test_multiindex_clean_names_method_chain(multiindex_dataframe):
+    df = jn.DataFrame(multiindex_dataframe).clean_names()
+
+    levels = [
+        ['a', 'bell_chart', 'decorated_elephant'],
+        ['b', 'normal_distribution', 'r_i_p_rhino_']
+    ]
+
+    labels = [[0, 1, 2], [0, 1, 2]]
+
+    expected_columns = pd.MultiIndex(levels=levels, labels=labels)
+    assert set(df.columns) == set(expected_columns)
+
+
+def test_multiindex_clean_names_pipe(multiindex_dataframe):
+    df = multiindex_dataframe.pipe(clean_names)
+
+    levels = [
+        ['a', 'bell_chart', 'decorated_elephant'],
+        ['b', 'normal_distribution', 'r_i_p_rhino_']
+    ]
+
+    labels = [[0, 1, 2], [0, 1, 2]]
+
+    expected_columns = pd.MultiIndex(levels=levels, labels=labels)
+    assert set(df.columns) == set(expected_columns)
```

Comment:
Wow, you really went above-and-beyond here! Thanks for doing this!

Reply:
Thanks - My only thought on this was, should we be handling `levels`, `labels` and maybe `expected_columns` at a global level so that I am not just repeating the setup of variables? I am not experienced with writing test so I am not sure what is good or standard practice.